lighthouse/classes/biosero.py

```diff
@@ -0,0 +1,77 @@
+import logging
+
+from lighthouse.classes.automation_system import AutomationSystem
+from lighthouse.classes.events.biosero import (
+    DestinationCreated,
+    DestinationFailed,
+    SourceAllNegatives,
+    SourceCompleted,
+    SourceNoPickableSamples,
+    SourceNoPlateMapData,
+    SourceUnrecognised,
+)
+from lighthouse.classes.events.biosero.source_partial import SourcePartial
+
+logger = logging.getLogger(__name__)
+
+
+class Biosero(AutomationSystem):
+    ###
+    # Events: https://ssg-confluence.internal.sanger.ac.uk/display/PSDPUB/%5BBiosero%5D+Cherrypicking+Events
+    ###
+    # Destination plate has been created successfully
+    EVENT_DESTINATION_CREATED = "lh_biosero_cp_destination_created"
+    # Destination plate failed to be created successfully
+    EVENT_DESTINATION_FAILED = "lh_biosero_cp_destination_failed"
+    # Destination plate has been partially filled
+    EVENT_DESTINATION_PARTIAL = "lh_biosero_cp_destination_partial"
+    # Source plate only contains negatives, nothing to cherrypick, and the plate is put into the output stacks
+    EVENT_SOURCE_ALL_NEGATIVES = "lh_biosero_cp_source_all_negatives"
+    # Source plate has had all pickable wells cherrypicked into destination plate(s)
+    EVENT_SOURCE_COMPLETED = "lh_biosero_cp_source_completed"
+    # Source plate has no pickable sample wells
+    EVENT_SOURCE_NO_PICKABLE_SAMPLES = "lh_biosero_cp_source_no_pickable_samples"
+    # Source plate has no related plate map data, cannot be cherrypicked (yet)
+    EVENT_SOURCE_NO_PLATE_MAP_DATA = "lh_biosero_cp_source_no_plate_map_data"
+    # Source plate has been partially picked
+    EVENT_SOURCE_PARTIAL = "lh_biosero_cp_source_partial"
+    # Source plate barcode cannot be read (damaged or missing), and the plate is put into the output stack
+    EVENT_SOURCE_UNRECOGNISED = "lh_biosero_cp_source_plate_unrecognised"
+
+    # needs to be an immutable object: https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables
+    PLATE_EVENT_NAMES = (
+        EVENT_DESTINATION_CREATED,
+        EVENT_DESTINATION_FAILED,
+        EVENT_DESTINATION_PARTIAL,
+        EVENT_SOURCE_ALL_NEGATIVES,
+        EVENT_SOURCE_COMPLETED,
+        EVENT_SOURCE_NO_PICKABLE_SAMPLES,
+        EVENT_SOURCE_NO_PLATE_MAP_DATA,
+        EVENT_SOURCE_PARTIAL,
+        EVENT_SOURCE_UNRECOGNISED,
+    )
+
+    def __init__(self) -> None:
+        self._name = AutomationSystem.AutomationSystemEnum.BIOSERO.name
+
+        self._event_destination_created = DestinationCreated(name=self.EVENT_DESTINATION_CREATED)
+        self._event_destination_failed = DestinationFailed(name=self.EVENT_DESTINATION_FAILED)
+        self._event_destination_partial = DestinationFailed(name=self.EVENT_DESTINATION_PARTIAL)
```

Comment:
Not sure I'd class partial destination as 'failed'. It can happen just because the system ran out of sources to process. As well as in various error scenarios.